packages/core/test/utils.spec.ts

```diff
@@ -33,9 +34,55 @@ describe('utils', () => {
     })
 
     it('should merge arrays', () => {
-      const target = [{ a: 'target' }, 'extraString'] as any
-      const source = [{ b: 'source' }] as any
-      expect(combine(target, source)).toEqual([{ a: 'target', b: 'source' }, 'extraString'])
+      const sourceA = [{ a: 'source a' }, 'extraString'] as any
+      const sourceB = [{ b: 'source b' }] as any
+      expect(combine(sourceA, sourceB)).toEqual([{ a: 'source a', b: 'source b' }, 'extraString'])
+    })
+
+    it('should merge multiple objects', () => {
+      expect(combine({ a: 1 }, { b: 2 }, { c: 3 })).toEqual({ a: 1, b: 2, c: 3 })
+    })
+
+    it('should not keep references on objects', () => {
+      const source = { a: { b: 1 } }
+      const result = combine({}, source)
+      source.a.b = 2
+      expect(result.a.b).toBe(1)
+    })
+
+    it('should not keep references on arrays', () => {
+      const source = { a: [1] }
+      const result = combine({}, source)
```

Comment:
```suggestion
      const result = combine(source)
```
or is there a reason to keep this one?

Reply:
* `combine` expects two arguments, so we can't really remove the first `{}`. We could change its signature to accept only one argument but it looks weird to have a `combine` call with only one argument (how is it combining?)
* I think we should keep this test case: it highlights that both objects and arrays are cloned.